kv/store_index.go

```diff
@@ -75,29 +74,97 @@ func (s *IndexStore) DeleteEnt(ctx context.Context, tx Tx, ent Entity) error {
 }
 
 // Find provides a mechanism for looking through the bucket via
-// the set options.
+// the set options. When a prefix is provided, the prefix is used with
+// the index, and not the entity bucket. If you wish to look at the entity
+// bucket and seek, then nest into the EntStore here and do that instead.
 func (s *IndexStore) Find(ctx context.Context, tx Tx, opts FindOpts) error {
 	span, ctx := tracing.StartSpanFromContext(ctx)
 	defer span.Finish()
 
-	if len(opts.Prefix) > 0 {
-		entCaptureFn := opts.CaptureFn
-		opts.CaptureFn = func(key []byte, indexVal interface{}) error {
-			ent, err := s.IndexStore.DecodeToEntFn(key, indexVal)
-			if err != nil {
-				return errors.Wrap(err, "index lookup")
-			}
+	if len(opts.Prefix) == 0 {
+		return s.EntStore.Find(ctx, tx, opts)
+	}
 
-			entVal, err := s.EntStore.FindEnt(ctx, tx, ent)
-			if err != nil {
-				return errors.Wrap(err, "entity lookup")
+	entCaptureFn := opts.CaptureFn
+	kvStream, filterFn := s.indexFilterStream(ctx, tx, opts.FilterEntFn)
+	opts.FilterEntFn = filterFn
+	opts.CaptureFn = func(key []byte, indexVal interface{}) error {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
This error needs to be wrapped.

Reply:
the errors get wrapped by the Find cmd in the storage base, extra wrapping here isn't warranted